src/TMap.cpp

```diff
@@ -1028,8 +1028,30 @@ bool TMap::findPath(int from, int to)
     return false;
 }
 
-bool TMap::serialize(QDataStream& ofs)
+bool TMap::serialize(QDataStream& ofs, int saveVersion)
 {
+    // clamp version values
+    if (saveVersion < 0) {
+        saveVersion = 0;
+    } else if (saveVersion > mMaxVersion) {
+        saveVersion = mMaxVersion;
+        QString errMsg = tr("[ ERROR ] -  The format {%1} you are trying to save the map with is too new\n"
```

Comment:
Sorry for making this difficult, but by `nil + msg` I meant return nil and the error message to Lua - it's not going to help the code if the message is printed to the user, code won't see it.

Reply:
I will need more input on this one. That's a behavior I haven't seen outside the interpreter and the only way I see it working nicely would be like `bool TMap::importMap(QFile& file, QString* errMsg)` 
and `bool TConsole::importMap(const QString& location, QString* errMsg)` do it. 

and then `if !console->saveMap() { lua_pushnil; lua_pushstrin(errMsg) }`
at the interpreter level.

You think there is a better way?

Reply:
Is this where you need an internal function to return success or an error msg? Have a look at how we used `std::pair` or `QPair` for that in the code.